docs/deployment/debian_ubuntu.md

```diff
@@ -0,0 +1,236 @@
+
+:::warning
+
+Without this, your program will crash due to missing libraries!
+
+:::
+
+Example of `your_program` starter script, setting the variable in a command-scope:
+
+```sh
+#!/bin/bash
+LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib/your_program your_program_executable
+```
+
+### 4) Make the desktop shortcut
+
+The desktop shortcut file follows the [freedesktop specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys). Arch Linux Wiki also has [good related information](https://wiki.archlinux.org/title/Desktop_entries).
+
+Below is an example of a desktop shortcut file:
+
+```
+[Desktop Entry]
+Name=YourProgram
+Comment=YourProgram, great for doing X
+Icon=your_program
+Exec=your_program
+Terminal=false
+Type=Application
+Categories=Development
+GenericName=YourProgram
+Keywords=keyword1; keyword2; keyword3
+```
+
+### 5) Add hicolor icons (optional)
+
+Hicolor icons follow a folder structure like below.
+
+[This blog post](https://martin.hoppenheit.info/blog/2016/where-to-put-application-icons-on-linux/) advises us to put icons on both `hicolor` and `pixmaps` directories, according to [Debian Menu System docs](https://www.debian.org/doc/packaging-manuals/menu.html/ch3.html#s3.7) and [FreeDesktop docs](https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-0.11.html#install_icons).
+
+```
+├── icons
+│   └── hicolor
+│       ├── 128x128
+│       │   └── apps
+│       │       └── your_program.png
+│       ├── 16x16
+│       │   └── apps
+│       │       └── your_program.png
+│       ├── 256x256
+│       │   └── apps
+│       │       └── your_program.png
+│       ├── 32x32
+│       │   └── apps
+│       │       └── your_program.png
+│       ├── 48x48
+│       │   └── apps
+│       │       └── your_program.png
+│       ├── 512x512
+│       │   └── apps
+│       │       └── your_program.png
+│       ├── 64x64
+│       │   └── apps
+│       │       └── your_program.png
+│       └── scalable
+│           └── apps
+│               └── your_program.svg
+```
+
+### 6) Compile the `.deb` package
+
+```sh
+# for x64 architectures, the suggested suffix is amd64.
+dpkg-deb --root-owner-group --build "./staging_folder/" "./your_program_${versionName}_amd64.deb"
+```
+
+## Example of a PowerShell script for the entire process
+
+```powershell
+$publishedProgramFilesDir = "./out" # directory that has the output of dotnet publish
+$stagingDir = "staging_folder"
+$versionName = "3.1.0"
+
+# Dotnet publish
+# PublishSingleFile is recommended to have less files
+# self-contained is recommended, so final users won't need to install .NET
+dotnet publish ./src/YourProgram.Desktop/YourProgram.Desktop.csproj `
+		--verbosity quiet `
+		--nologo `
+		--configuration Release `
+		-p:PublishSingleFile=true ` 
```

Comment:
Technically this is not recommended, since it can and will break certain code patterns, like ones relying on `typeof(Foo).Assembly.Locations`